File: 2021/test/day_04_test.rb
# frozen_string_literal: true

require "test_helper"
require "day_04"

class Day04Test < Minitest::Test
  def test_part1
    day = Day04.new

    assert_equal 4512, day.part1(input("day_04_example_1"))
    assert_equal 65_325, day.part1(input("day_04"))
  end

  def test_part2
    day = Day04.new

    assert_equal 1924, day.part2(input("day_04_example_1"))
    assert_equal 4624, day.part2(input("day_04"))
  end
end

File: 2025/test/day_04_test.exs
defmodule Day04Test do
  use ExUnit.Case

  import TestHelper

  test "part1" do
    assert Day04.part_1(input("day_04_example_1")) == 13
    assert Day04.part_1(input("day_04")) == 1551
  end

  test "part2" do
    # assert Day04.part_2(input("day_04_example_1")) == 3_121_910_778_619
    # assert Day04.part_2(input("day_04")) == 171_846_613_143_331
  end
end
